File: C#/3 - Software design patterns/QABank/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QABank
{
    public abstract class Account
    {
        protected int balance;
        protected string name;

        public abstract bool Withdraw(int amount);

        public int Balance
        {
            get
            {
                return this.balance;
            }
        }

        public string Name
        {
            get
            {
                return this.name;
            }
        }

        public bool Deposit(int amount)
        {
            this.balance += amount;
            return true;
        }

        public override string ToString()
        {
            return $"BALANCE={Balance}, NAME={Name}";
        }
    }
}

File: Java/3 - Software design patterns/src/main/java/com/qaicpads/Account.java
package com.qaicpads;

public abstract class Account {
	
	protected int balance;
	protected String name;

	public abstract boolean withdraw(int amt);
	
	public int getBalance() {
		return this.balance;
	}
	
	public String getName() {
		return this.name;
	}
	
	
	@Override
	public String toString(){
		return "BALANCE="+this.getBalance()+", NAME="+this.getName();
	}
	
	public boolean deposit(int amt) {
		this.balance += amt;
		return true;
	}

}
